src/TicTacToe.ConsoleApp/GameEngine.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using TicTacToe.ConsoleApp.Configuration;
+using TicTacToe.Models;
+using TicTacToe.Services;
+using TicTacToe.Services.Exceptions;
+using TicTacToe.Services.Interfaces.Models;
+
+namespace TicTacToe.ConsoleApp
+{
+    public class GameEngine
+    {
+        public void PlayGame(Guid gameId, Guid userId)
+        {
+            while (true)
+            {
+                using (var context = new TicTacToeDbContextFactory().CreateDbContext())
+                {
+                    var gameService = new GameService(context, new GameResultValidator());
+
+                    var game = gameService.Status(gameId, userId);
+                    this.PrintBoard(game.Board);
+
+                    if (game.State == GameState.WaitingForASecondPlayer)
+                    {
```

Comment:
Separate the __if-else__ checks into a new method to improve the method readability and to minimize the complexity. Name the method, for example, `PrintState(game, userId, gameService)` and call it under the `this.PrintBoard(game.Board);`

Reply:
done